modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rule/logical/LogicalOrToUnionRule.java

```diff
@@ -155,6 +212,14 @@ private RelNode createUnionAll(RelOptCluster cluster, RelNode input, RexNode op1
             .build();
     }
 
+    /** */
+    private static boolean preMatch(IgniteLogicalTableScan scan) {
+        return scan.condition() != null &&
+            // _key_PK not interesting here, but it`s depend on current PK implementation, in future PK can be removed
+            // and this condition will become incorrect.
+            scan.getTable().unwrap(IgniteTable.class).indexes().size() >= 2;
```

Comment:
honestly, I would prefer to omit this check just because it could be easily broken, and no one will ever noticed that. AFAIU, it doesn't prune a search space, but allows to avoid unnecessary computations only.

Reply:
i`m ok with your suggestion, lets remove it.